### ISSUE TITLE
Gradients of a layer output w.r.t. input as input to new model

### ISSUE THREAD

**Issue by jecaJeca** [keras-team/keras#13087]
I am doing research and for an experiment I want to use gradients of a specific layer in the network with respect to the network's input (similar as guided backprop) as input to another network (classifier). The goal is to 'force' network to change 'attention' according to classifier, so those two networks should be trained simultaneously. 

Is something like this possible to do in Keras? I was suggested to switch to Tensorflow, but because of the rest of the project it would be very nice if I can stay in Keras.  

I tried to implement on this way : 

    input_tensor = model.input
    output_tensor = model.layers[-2].output
    grad_calc = keras.layers.Lambda(lambda x:K.gradients(x,input_tensor)[0],output_shape=(256,256,3),trainable=False)(output_tensor)
    pred = classifier(grad_calc)
    out_model = Model(input_tensor,pred)
    out_model.compile(loss='mse',optimizer=keras.optimizers.Adam(0.0001),metrics=['accuracy'])

Then, when I try to train the model 

    out_model.train_on_batch(imgs,np.zeros((imgs.shape[0],2)))

it is not working. It seems that it stucks there, nothing is happening (no error nor other message).

I am not sure is this the right way to approach this problem, so I would be very thankful if someone with more experience can take a look and give me advice.

**Comment by jecaJeca** [comment]
The problem with previous model was in BatchNormalisation layer. It seems that it caused infinite recursion.
